packages/react/src/xcss-prop/index.ts

```diff
@@ -1,27 +1,30 @@
 import type * as CSS from 'csstype';
 
+import type { ApplySchemaValue } from '../create-strict-api/types';
 import { ac } from '../runtime';
-import type { CSSPseudos, CSSProperties, StrictCSSProperties } from '../types';
+import type { CSSPseudos, CSSPseudoClasses, CSSProperties, StrictCSSProperties } from '../types';
 
 type MarkAsRequired<T, K extends keyof T> = T & { [P in K]-?: T[P] };
 
-type XCSSItem<TStyleDecl extends keyof CSSProperties, TSchema> = {
-  [Q in keyof CSSProperties]: Q extends TStyleDecl
-    ?
-        | CompiledPropertyDeclarationReference
-        | (Q extends keyof TSchema ? TSchema[Q] : CSSProperties[Q])
+type XCSSValue<
+  TStyleDecl extends keyof CSSProperties,
+  TSchema,
+  TPseudoKey extends CSSPseudoClasses | ''
+> = {
+  [Q in keyof StrictCSSProperties]: Q extends TStyleDecl
+    ? CompiledPropertyDeclarationReference | ApplySchemaValue<TSchema, Q, TPseudoKey>
```

Comment:
We re-use the `ApplySchemaValue` here from the strict types, using the same type behaviour to get the value.

Reply:
I feel like something in here isn't working as I expected given `xcss={styles.primary}` passes even when the type does not match `TSchema` and the `cssMap` fails 😕 but I can't place my finger on it…

Reply:
Yeah I didn't spend that much time inside xcss prop types. I got the type violations fixed and then moved on... What's your (deeper) thoughts?

Reply:
Oh, I think this introduces a bug 😢 using `import { cssMap } from '@compiled/react'` I can now pass anything I want.

<img width="841" alt="Screenshot 2024-03-05 at 2 35 11 PM" src="https://github.com/atlassian-labs/compiled/assets/136543114/2a22aec8-1ea0-48af-9786-775114a46a8a">

Can't find a solution for it, I'll try to look a bit later today if I can..